lib/internal/queue_microtask.js

```diff
@@ -0,0 +1,31 @@
+'use strict';
+
+const { ERR_INVALID_ARG_TYPE } = require('internal/errors').codes;
+const { AsyncResource } = require('async_hooks');
+const { getDefaultTriggerAsyncId } = require('internal/async_hooks');
+const { enqueueMicrotask } = process.binding('util');
```

Comment:
`internalBinding`?

That being said… could the implementation use `Promise`s to avoid calling into C++ altogether?

Reply:
we could also use `%EnqueueMicrotask` in v8, and fall back to the c++ call for chakra/whatever else